auth.js

```diff
@@ -0,0 +1,26 @@
+const express = require('express');
+const passport = require('passport');
+const router = express.Router();
+
+const snapchatStrategy = require('./strategies/snapchat');
+
+function successHandler(req, res) {
+    console.log('Successful authentication');
+};
+
+function errorHandler(er, req, res, next) {
+    console.log('Failed to authenticate');
```

Comment:
if you [throw](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw) an error, the user will be able to `catch` it. ie
```js
// server
function errorHandler(er, req, res, next) {
  throw new Error('Failed to authenticate');
}

// client 
tryingToAuthenticate()
  .then(res => {})
  .catch(er => {
     console.debug('Yay I caught it')
  });
```

Reply:
seems like express also has another error catching method `next(err)` for errors from an async function ([ref](http://expressjs.com/en/guide/error-handling.html))

Reply:
cool